ci_feature/main.sh

```diff
@@ -196,16 +174,94 @@ if [ "$cAdvisorIsSecure" = true ] ; then
       echo "export IS_SECURE_CADVISOR_PORT=true" >> ~/.bashrc
       export CADVISOR_METRICS_URL="https://$NODE_IP:10250/metrics"
       echo "export CADVISOR_METRICS_URL=https://$NODE_IP:10250/metrics" >> ~/.bashrc
+      echo "Making wget request to cadvisor endpoint /pods with port 10250 to get the configured container runtime on kubelet"
+      podsResponse=$(wget -O- --server-response https://$NODE_IP:10250/pods --no-check-certificate --header="Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)")
+      if [ -z "$podsResponse" ]; then
+            echo "-e error  wget request to cadvisor endpoint /pods with port 10250 to get the configured container runtime on kubelet failed"
+            # should we default to container runtime docker?
+      else
+            containerRuntime=$(echo $podsResponse | jq -r '.items[0].status.containerStatuses[0].containerID' | cut -d ':' -f 1)
+            nodeName=$(echo $podsResponse | jq -r '.items[0].spec.nodeName')
+            export CONTAINER_RUN_TIME=$containerRuntime
+            export NODE_NAME=$nodeName
+            echo "configured container runtime on kubelet is : "$CONTAINER_RUN_TIME
+            echo "export CONTAINER_RUN_TIME="$CONTAINER_RUN_TIME >> ~/.bashrc
+            echo "export NODE_NAME="$NODE_NAME >> ~/.bashrc
+      fi
+
 else
       echo "Wget request using port 10250 failed. Using port 10255"
       export IS_SECURE_CADVISOR_PORT=false
       echo "export IS_SECURE_CADVISOR_PORT=false" >> ~/.bashrc
       export CADVISOR_METRICS_URL="http://$NODE_IP:10255/metrics"
       echo "export CADVISOR_METRICS_URL=http://$NODE_IP:10255/metrics" >> ~/.bashrc
+      echo "Making wget request to cadvisor endpoint with port 10255 to get the configured container runtime on kubelet"
+      podsResponse=$(wget -O- --server-response http://$NODE_IP:10255/pods)
+      if [ -z "$podsResponse" ]; then
+            echo "-e error  wget request to cadvisor endpoint /pods with port 10250 to get the configured container runtime on kubelet failed"
+            # should we default to container runtime docker?
+      else
+            containerRuntime=$(echo $podsResponse | jq -r '.items[0].status.containerStatuses[0].containerID' | cut -d ':' -f 1)
```

Comment:
What is podsResponse doesnt have valid response body?

Reply:
addressed.